common/component/azure/eventhubs/eventhubs.go

```diff
@@ -295,15 +298,29 @@ func (aeh *AzureEventHubs) Subscribe(subscribeCtx context.Context, config Subscr
 				}
 			}()
 		}
-	}()
+	}
+	go subscriberLoop()
 
 	// Start the processor
 	go func() {
-		// This is a blocking call that runs until the context is canceled
-		err = processor.Run(subscribeCtx)
-		// Do not log context.Canceled which happens at shutdown
-		if err != nil && !errors.Is(err, context.Canceled) {
-			aeh.logger.Errorf("Error from event processor: %v", err)
+		for {
+			// This is a blocking call that runs until the context is canceled
+			err = processor.Run(subscribeCtx)
+			// Do not log context.Canceled which happens at shutdown
+			if err != nil && errors.Is(err, context.Canceled) {
+				return
+			} else {
+				if err != nil {
+					aeh.logger.Errorf("Error from event processor: %v", err)
+				} else {
+					aeh.logger.Debugf("Event processor terminated without error")
+				}
+				// wait for subscription loop finished signal
+				<-subscriptionLoopFinished
+				// wait for 5 seconds before restarting the subscription loop
+				<-time.After(5 * time.Second)
```

Comment:
2 things:

1. Please use a backoff that is exponential (upon success, it should be reset)
2. If the context is canceled, we should stop waiting and return

See how we do here: https://github.dev/dapr/components-contrib/blob/ba6609e624e0404e03d75ccffda6e5de8d94267b/bindings/azure/servicebusqueues/servicebusqueues.go#L145-L155